test/app-tap/gh-2717-no-quit-sigint.test.lua

```diff
@@ -0,0 +1,201 @@
+
+ph_client:write('require("console").connect(\'unix/:' .. sock .. '\')\n')
+
+local client_data = ''
+while string.endswith(client_data, 'unix/:' .. sock .. '>') == nil
+        and clock.monotonic() - start_time < deadline do
+    local cur_data = ph_client:read({timeout = 3.0})
+    if cur_data ~= nil and cur_data ~= '' then
+        client_data = client_data .. cur_data
+    end
+end
+assert(clock.monotonic() - start_time < deadline, 'deadline is over')
+
+ph_client:signal(popen.signal.SIGINT)
+test:unlike(ph_client:info().status.state, popen.state.EXITED,
+            'SIGINT doesn\'t kill nested tarantool in interactive mode')
+while string.endswith(client_data, 'C\n---\n...\n\nunix/:' .. sock .. '>') == nil
+        and clock.monotonic() - start_time < deadline do
+    local cur_data = ph_client:read({timeout = 3.0})
+    if cur_data ~= nil and cur_data ~= '' then
+        client_data = client_data .. cur_data
+    end
+end
+
+ph_server:signal(popen.signal.SIGINT)
+
+local status = ph_server:wait(nil, popen.signal.SIGINT)
+test:unlike(status.state, popen.state.ALIVE,
+            'SIGINT terminates tarantool in daemon mode')
+
+start_time = clock.monotonic()
+local data = ''
+while data:match('got signal 2') == nil
+        and clock.monotonic() - start_time < deadline do
+    data = data .. process_timeout.read_with_timeout(f,
+            file_read_timeout,
+            file_read_interval)
+end
+assert(data:match('got signal 2'), 'there is no one note about SIGINT')
+assert(clock.monotonic() - start_time < deadline, 'deadline is over')
+
+f:close()
+ph_server:close()
+ph_client:close()
+os.remove(log_file)
+os.remove(pid_file)
+os.remove(xlog_file)
+os.remove(snap_file)
+
+--
+-- Testing case when the client and instance are called in the same console.
+--
+cmd = 'ERRINJ_STDIN_ISATTY=1 ' .. TARANTOOL_PATH .. ' -i 2>&1'
+ph = popen.new({cmd}, {
+    shell = true,
+    setsid = true,
+    group_signal = true,
+    stdout = popen.opts.PIPE,
+    stderr = popen.opts.DEVNULL,
+    stdin = popen.opts.PIPE,
+})
+assert(ph, 'process is not up')
+
+start_time = clock.monotonic()
+deadline = 10.0
+
+output = ''
+while output:find(prompt) == nil
+        and clock.monotonic() - start_time < deadline do
+    data = ph:read({timeout = 1.0})
+    if data ~= nil then
+        output = output .. data
+    end
+end
+assert(clock.monotonic() - start_time < deadline, 'deadline is over')
+
+prompt = 'unix/:' .. sock .. '> '
+ph:write('_ = require(\'console\').listen(\'' .. sock .. '\')\n',
+         {timeout = 1.0})
+ph:write('_ = require(\'console\').connect(\'' .. sock .. '\')\n',
+         {timeout = 1.0})
+
+fiber.sleep(0.2)
+ph:signal(popen.signal.SIGINT)
+
+start_time = clock.monotonic()
+while string.endswith(output, prompt .. '^C\n---\n...\n\n' .. prompt) == false
+        and clock.monotonic() - start_time < deadline do
+    local data = ph:read({timeout = 1.0})
+    if data ~= nil then
+        output = output .. data
+    end
+end
```

Comment:
Almost all such blocks end with an assertion regarding spent time. Should we assert after this one as well?

Reply:
Yes, I've missed this place. Fixed.